project/src/UI/Upgrades/UpgradeButton.gd

```diff
@@ -3,23 +3,84 @@
 tool
 extends TextureButton
 
+signal on_ani_complete(is_entry)
+
 export var texture: Texture setget set_texture
 export var text := "" setget set_text
 
 onready var texture_rect := $VBoxContainer/TextureRect
 onready var label := $VBoxContainer/Label
 
+onready var tween := $Tween
+
+onready var on_pos = self.rect_position
+onready var off_pos = self.rect_position + Vector2(0, 200);
+
+onready var is_entry_ani = true;
+
+
+func animate(entry: bool, delay: float) -> void:
+	self.is_entry_ani = entry;
+	var start_pos = off_pos if entry else on_pos
+	var end_pos = on_pos if entry else off_pos
+	var start_scale = Vector2.ZERO if entry else Vector2.ONE
+	var end_scale = Vector2.ONE if entry else Vector2.ZERO
+	var start_color = Color.transparent if entry else Color.white
+	var end_color = Color.white if entry else Color.transparent
+	var easing = Tween.EASE_OUT if entry else Tween.EASE_IN
+	var fade_time = 0.5 if entry else 0.7
+	self.rect_position = start_pos
+	self.rect_scale = start_scale
+	self.modulate = start_color
+
+	tween.interpolate_property(
+		self,
+		'rect_position',
+		start_pos,
+		end_pos,
+		1.0,
+		Tween.TRANS_EXPO,
+		easing,
+		delay
+	)
+	tween.interpolate_property(
+		self,
+		'rect_scale',
+		start_scale,
+		end_scale,
+		1.2,
+		Tween.TRANS_ELASTIC,
+		easing,
+		delay
+	)
+	tween.interpolate_property(
+		self,
+		'modulate',
+		start_color,
+		end_color,
+		fade_time,
+		Tween.TRANS_LINEAR,
+		Tween.TRANS_LINEAR,
+		delay
+	)
+	tween.start()
+
 
 func set_texture(value: Texture) -> void:
 	texture = value
 	if not texture_rect:
 		yield(self, "ready")
 	texture_rect.texture = value
 
-
+	
 func set_text(value: String) -> void:
 	text = value
 	if not texture_rect:
 		yield(self, "ready")
 	label.text = value
 	label.visible = text != ""
+
+	
+# Animation tween callback signal
+func _on_all_completed():
```

Comment:
The guideline for signal callbacks is _on_Node_signal_name, here, `_on_Tween_tween_all_completed`. With a clear name like that, you don't need the extra comment.